aten/src/ATen/DeviceAccelerator.cpp

```diff
@@ -18,6 +18,8 @@ C10_API std::optional<DeviceType> getAccelerator(bool checked) {
     } else if (at::hasCUDA()) {
         CHECK_NO_PU1
         return kCUDA;
+    } else if (at::hasMTIA()) {
```

Comment:
Can you add a CHECK_NO_MTIA macro and add it to the other conditions?
Also add NO_CUDA/NO_PU1 to this branch